### PR TITLE
Block requests do not exceed MAX_REQUEST_BLOCKS.

### PR DESCRIPTION
**Pull request by g-r-a-n-t**
### What was wrong?

Block requests were not being made to recently connected peers on Altona, since the number of blocks needed to sync exceeded the maximum amount blocks that can be requested. 

### How was it fixed?
The chain is synced by making multiple consecutive requests of size <= `MAX_REQUEST_BLOCKS`.

*Note: Under abnormal network condition or after some rounds of BeaconBlocksByRange requests, the client might need to send Status request again to learn if the peer has a higher head. Implementers are free to implement such behavior in their own way.*

^ the issue of re-exchanging statuses after having finally synced to the peers head is not resolved in this PR. It may be worth revisiting later, once we're at a point where we can sync blocks fast enough to catch the initial head.

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] Clean up commit history

[//]: # (For important changes that should go into the release notes please add a newsfragment file as explained here: https://github.com/ethereum/trinity/blob/master/newsfragments/README.md)

[//]: # (See: https://trinity-client.readthedocs.io/en/latest/contributing.html#pull-requests)
- [ ] Add entry to the [release notes](https://github.com/ethereum/trinity/blob/master/newsfragments/README.md)

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

**Comment by ralexstokes** [comment]
> ^ the issue of re-exchanging statuses after having finally synced to the peers head is not resolved in this PR. It may be worth revisiting later, once we're at a point where we can sync blocks fast enough to catch the initial head.

yeah in my mind clients should just regularly check some (even all?) of their peers for their latest head to have as wide a view of the network state as possible. we could imagine the final step to an initial sync is to "double-check" that the peer reports their head as ours -- want to move this to an issue so we can track it?

**Comment by g-r-a-n-t** [comment]
Added :+1:

File: trinity/nodes/beacon/full.py [hunk 1]
```diff
@@ -388,25 +388,26 @@ async def _handle_block_gossip(self) -> None:
...
+        )
+        requests = []
+        while finalized_slot < status.head_slot:
+            count = min(status.head_slot - finalized_slot, MAX_REQUEST_BLOCKS)
+            requests.append(SyncRequest(
+                peer_id, Slot(finalized_slot + 1), count
+            ))
+            finalized_slot += count
+
+        return tuple(requests)
```
Comment (ralexstokes) [inline trinity/nodes/beacon/full.py:410]: also worth noting we have a `to_tuple` decorator (e.g. in `eth_utils` package) that we use frequently...

File: trinity/nodes/beacon/full.py [hunk 2]
```diff
@@ -388,25 +388,26 @@ async def _handle_block_gossip(self) -> None:
...
+            head_state.finalized_checkpoint.epoch,
+            self._eth2_config.SLOTS_PER_EPOCH
+        )
+        requests = []
+        while finalized_slot < status.head_slot:
+            count = min(status.head_slot - finalized_slot, MAX_REQUEST_BLOCKS)
+            requests.append(SyncRequest(
+                peer_id, Slot(finalized_slot + 1), count
+            ))
+            finalized_slot += count
```
[the remark this answers is not in the document]
Comment (g-r-a-n-t) [reply trinity/nodes/beacon/full.py:408]: Doing some quick calculations, the number of iterations would increase by `365 * 24 * 60 * 60 / (12 * 1024)` (2566) per year, which shouldn't be a problem as long as we don't add any expensive operations to the loop's body.

**Pull request merged**